crates/iceberg/src/spec/values.rs

```diff
@@ -0,0 +1,663 @@
+                "Failed to convert microseconds to time",
+            ))?,
+            Utc,
+        ))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    #[test]
+    pub fn boolean() {
+        let input = Value::Boolean(true);
+
+        let raw_schema = r#""boolean""#;
+
+        let schema = apache_avro::Schema::parse_str(raw_schema).unwrap();
+
+        let mut writer = apache_avro::Writer::new(&schema, Vec::new());
+
+        writer.append_ser(input.clone()).unwrap();
+
+        let encoded = writer.into_inner().unwrap();
+
+        let reader = apache_avro::Reader::new(&*encoded).unwrap();
+
+        for record in reader {
+            let result = apache_avro::from_value::<Value>(&record.unwrap()).unwrap();
+            assert_eq!(input, result);
+        }
+    }
+
+    #[test]
+    pub fn int() {
+        let input = Value::Int(42);
+
+        let raw_schema = r#""int""#;
+
+        let schema = apache_avro::Schema::parse_str(raw_schema).unwrap();
+
+        let mut writer = apache_avro::Writer::new(&schema, Vec::new());
+
+        writer.append_ser(input.clone()).unwrap();
+
+        let encoded = writer.into_inner().unwrap();
+
+        let reader = apache_avro::Reader::new(&*encoded).unwrap();
+
+        for record in reader {
+            let result = apache_avro::from_value::<Value>(&record.unwrap()).unwrap();
+            assert_eq!(input, result);
+        }
+    }
+
+    #[test]
+    pub fn float() {
+        let input = Value::Float(OrderedFloat(42.0));
+
+        let raw_schema = r#""float""#;
+
+        let schema = apache_avro::Schema::parse_str(raw_schema).unwrap();
+
+        let mut writer = apache_avro::Writer::new(&schema, Vec::new());
+
+        writer.append_ser(input.clone()).unwrap();
+
+        let encoded = writer.into_inner().unwrap();
+
+        let reader = apache_avro::Reader::new(&*encoded).unwrap();
+
+        for record in reader {
+            let result = apache_avro::from_value::<Value>(&record.unwrap()).unwrap();
+            assert_eq!(input, result);
+        }
+    }
+
+    #[test]
+    pub fn string() {
+        let input = Value::String("test".to_string());
+
+        let raw_schema = r#""string""#;
+
+        let schema = apache_avro::Schema::parse_str(raw_schema).unwrap();
+
+        let mut writer = apache_avro::Writer::new(&schema, Vec::new());
+
+        writer.append_ser(input.clone()).unwrap();
+
+        let encoded = writer.into_inner().unwrap();
+
+        let reader = apache_avro::Reader::new(&*encoded).unwrap();
+
+        for record in reader {
+            let result = apache_avro::from_value::<Value>(&record.unwrap()).unwrap();
+            assert_eq!(input, result);
+        }
+    }
+
+    #[test]
+    pub fn struct_value() {
+        let input = Value::Struct(Struct::from_iter(vec![(
+            "name".to_string(),
+            Some(Value::String("Alice".to_string())),
+        )]));
+
+        let raw_schema = r#"{"type": "record","name": "r102","fields": [{
```

Comment:
I'm a bit confused here, are we serializing Avro or Iceberg schema here? This is an Avro schema. The Iceberg equivalent would be:

```python
Python 3.9.6 (default, May  7 2023, 23:32:44) 
[Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from pyiceberg.types import *
>>> StructType(NestedField(1, "name", StringType())).json()
'{"type": "struct", "fields": [{"id": 1, "name": "name", "type": "string", "required": true}]}'
```

When I extract the `partition` field `avro-tools`:
```
{
	"partition": {
		"tpep_pickup_datetime_day": {
			"int": 19088
		}
	}
}
```

So in Python, I went all the way. What I did there is implement my own Avro reader that takes the Iceberg schema as input. I'm not saying that you should that do here, but it would maybe be helpful to implement a visitor to convert an Iceberg schema to an Avro schema. I would say that the `partition` would read into a `Vec<Any>`